zookeeper-server/src/main/java/org/apache/zookeeper/jmx/MBeanRegistry.java

```diff
@@ -43,28 +43,26 @@
     public static final String DOMAIN = "org.apache.ZooKeeperService";
 
     private static final Logger LOG = LoggerFactory.getLogger(MBeanRegistry.class);
-    private static volatile MBeanRegistry instance = new MBeanRegistry();
 
     private final Object LOCK = new Object();
 
     private Map<ZKMBeanInfo, String> mapBean2Path = new ConcurrentHashMap<ZKMBeanInfo, String>();
 
     private MBeanServer mBeanServer;
 
-    /**
-     * Useful for unit tests. Change the MBeanRegistry instance
-     *
-     * @param instance new instance
-     */
-    public static void setInstance(MBeanRegistry instance) {
-        MBeanRegistry.instance = instance;
+    private static class MBeanRegistryHolder {
```

Comment:
Why do we need this additional class?

Reply:
@eolivelli This is just a singleton style a programmer prefers. Look at this:
[1] https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom#Example_Java_Implementation
[2] https://stackoverflow.com/questions/17799976/why-is-static-inner-class-singleton-thread-safe